session.c

```diff
@@ -2158,7 +2158,12 @@ session_signal_req(struct ssh *ssh, Session *s)
 
 	debug_f("signal %s, killpg(%ld, %d)", signame, (long)s->pid, sig);
 	temporarily_use_uid(s->pw);
+#ifndef __TANDEM
 	r = killpg(s->pid, sig);
+#else
+	/* NonStop does not implement killpg() */
+	r = kill(s->pid, sig);
```

Comment:
this should be done by providing a killpg implementation instead of modifying the main code.  I've just added one, please let me know if it doesn't work.